test/e2e/network/service.go

```diff
@@ -1261,22 +1268,30 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectNoError(err)
 		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, serverPod1.Name, f.Namespace.Name, framework.PodStartTimeout))
 
+		// Check that the pod receives the traffic
+		ginkgo.By(fmt.Sprintf("Sending UDP traffic to NodePort service %s on node with public IP %s", serviceName, nodeIP))
 		errorChannel := make(chan error)
 		signal := make(chan struct{}, 1)
 		go continuousEcho(nodeIP, int(udpService.Spec.Ports[0].NodePort), 3*time.Second, 20, signal, errorChannel)
 
+		// Create a second pod
 		ginkgo.By("creating a second pod for the service " + serviceName)
 		serverPod2 := newAgnhostPod(serverPod2Name, "netexec", fmt.Sprintf("--udp-port=%d", 80))
 		serverPod2.Labels = udpJig.Labels
 		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), serverPod2, metav1.CreateOptions{})
 		framework.ExpectNoError(err)
 		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, serverPod2.Name, f.Namespace.Name, framework.PodStartTimeout))
-		signal <- struct{}{}
 
+		// And delete the first one
 		framework.Logf("Cleaning up %s pod", serverPod1Name)
 		err = cs.CoreV1().Pods(ns).Delete(context.TODO(), serverPod1Name, metav1.DeleteOptions{})
 		framework.ExpectNoError(err, "failed to delete pod: %s on node", serverPod1Name)
 
+		// Check that the second pod keeps receiving traffic
+		ginkgo.By(fmt.Sprintf("Sending UDP traffic to NodePort service %s on node with public IP %s", serviceName, nodeIP))
+		signal <- struct{}{}
```

Comment:
reading the issue I think that we should start once the first pod is deleted, but you may have a better insights in the order, feel free to correct me